src/test/java/org/mule/extension/file/FileListTestCase.java

```diff
@@ -132,12 +133,11 @@ private boolean assertListedFiles(List<Message> messages) throws Exception {
   }
 
   private List<Message> doList(String flowName, String path, boolean recursive) throws Exception {
-    List<Message> messages =
-        (List<Message>) flowRunner(flowName).withVariable("path", path).withVariable("recursive", recursive).run()
+    Message messages =
+        (Message) flowRunner(flowName).withVariable("path", path).withVariable("recursive", recursive).run()
             .getMessage().getPayload().getValue();
-
-    assertThat(messages, is(notNullValue()));
-
-    return messages;
+    assertThat(messages.getPayload(), is(notNullValue()));
```

Comment:
this assertions are useless. If this is not true, the the test method consuming this will fail. What you should really test here is the data type of each of the returned messages